test/framework/src/main/java/org/elasticsearch/test/rest/client/RestTestResponse.java

```diff
@@ -82,6 +87,14 @@ public String getReasonPhrase() {
         return response.getStatusLine().getReasonPhrase();
     }
 
+    public Map<String, String> getHeaders() {
+        Map<String, String> headers = new HashMap<>();
+        for (Header header: response.getHeaders()) {
+            headers.put(header.getName(), header.getValue());
```

Comment:
nit: multiple headers coming from http client can theoretically have same name but different values. I think the correct way of representing them is a Map<String, List<String>>, otherwise only one value survives in our final map.